### PR TITLE
Ignore known annotations in parameter binding

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #88

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java [hunk 1]
```diff
@@ -581,6 +581,19 @@ private List<TemplateParameterInfo> gatherTemplateParamInfo(List<MethodParameter
         return templateParams;
     }
 
+    private List<AnnotationInstance> effectiveParamAnnotations(MethodParameterInfo param) {
+        return param.annotations().stream().filter(ai -> {
+            String name = ai.name().toString();
+            if (name.startsWith("kotlin")) {
+                return false;
+            }
+            if (name.endsWith("NotNull")) {
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:590]: I wonder if you should ignore everything under `jakarta.validation.constraints.` instead.

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:590]: NotNull are super common - most tools even allow custom ones, so I'll leave the rule here

Comment (cescoffier) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:590]: I was wondering if we shouldn't add CheckReturnValue.

Comment (geoand) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:590]: That is only for return values,  no?

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java:590]: Jakarta Validation also has `@NotBlank` which could be useful in this case. Thus my recommendation.

You can keep not null as is but I would also ignore everything under jakarta.validation.constraints.

**Pull request merged**